pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -406,25 +422,47 @@ func (p *ConfigurationEditFlags) Apply(
 		}
 	}
 
-	if cmd.Flags().Changed("annotation") {
-		annotationsMap, err := util.MapFromArrayAllowingSingles(p.Annotations, "=")
+	if cmd.Flags().Changed("annotation") || cmd.Flags().Changed("annotation-service") || cmd.Flags().Changed("annotation-revision") {
+		annotationsAllMap, err := util.MapFromArrayAllowingSingles(p.Annotations, "=")
 		if err != nil {
 			return fmt.Errorf("Invalid --annotation: %w", err)
 		}
+		annotationRevisionFlagMap, err := util.MapFromArrayAllowingSingles(p.AnnotationsRevision, "=")
+		if err != nil {
```

Comment:
I can comment out lines 431-433 and all tests pass

Reply:
Have refactored the tests